defs.bzl

```diff
@@ -32,59 +12,73 @@ def _source_files_in(ctx, attr):
 
     return [f for f in files if f.is_source]
 
-def _check_format(ctx, package, f):
+def _do_format(ctx, f, format_options, execution_reqs):
     binary = ctx.attr._binary.files_to_run.executable
-    dry_run = ctx.attr._dry_run[BuildSettingInfo].value
-
-    out = ctx.actions.declare_file(
-        "{name}.clang_format".format(
-            # don't duplicate package in the out file
-            name = f.short_path.removeprefix(package + "/"),
-        ),
-    )
+    out = ctx.actions.declare_file(f.short_path + ".clang_format")
 
-    ctx.actions.run(
-        inputs = [ctx.file._config] + ([binary] if binary else []) + [f],
+    ctx.actions.run_shell(
+        inputs = [
+            ctx.file._config,
+            f,
+        ] + ([binary] if binary else []),
         outputs = [out],
-        executable = ctx.executable._wrapper,
-        arguments = [
-            binary.path if binary else "clang-format",
-            ctx.file._config.path,
-            f.path,
-            out.path,
-        ] + (["--dry-run"] if dry_run else [""]),
+        command = """
+set -euo pipefail
+
+test -e .clang-format || ln -s -f {config} .clang-format
```

Comment:
3rd party should really avoid doing any symlinks especially using `-f` flag. Clang-tidy supports `clang-format -style=file:<path/to/config/file> ...` notation since 14.0.0.

For example one can do:
`clang-format -style=file:/home/username/projects/someproject/.clang-format -i main.cpp`

Would you fix that it might worth to mention minimal `clang-format` version on README site

Reply:
I forgot that sandboxing is disabled with the new fix aspect. Sorry!

Reply:
No problem :-) Should be very straightforward to fix that